perf_timer/src/arch.rs

```diff
@@ -8,8 +8,8 @@ pub use aarch64::Aarch64 as Arch;
 
 // QEMU uses the ACPI frequency when CPUID-based frequency determination is not available.
 const DEFAULT_ACPI_TIMER_FREQUENCY: u64 = 3579545;
-
 static PERF_FREQUENCY: AtomicU64 = AtomicU64::new(0);
+const PM_TIMER_PORT: u16 = 0x608; // Obtained through FADT's X_PM_TMR_BLK field

```

Comment:
I don't have time to look through the PR completely right now, but you'll need to make this ACPI Fixed Hardware base configurable from either an I/O port or an MMIO address in the final solution.